src/main.go

```diff
@@ -88,8 +89,17 @@ type query struct {
 }
 
 type Conn struct {
+	// DB might be a shared one by multiple Conn, if the connection information are the same.
 	mdb *sql.DB
-	tx  *sql.Tx
```

Comment:
The old code is buggy because it use a sql.Tx to run the query, and if tx is nil, we have to create one, so the logic become "begin; query; commit"